server/sonar-server/src/main/java/org/sonar/server/startup/FeedUsersLocalStartupTask.java

```diff
@@ -0,0 +1,115 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.startup;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.picocontainer.Startable;
+import org.sonar.api.config.Settings;
+import org.sonar.api.user.UserQuery;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.loadedtemplate.LoadedTemplateDto;
+import org.sonar.db.user.UserDto;
+import org.sonar.server.user.UserUpdater;
+
+import static java.util.Arrays.asList;
+import static org.sonar.api.CoreProperties.CORE_AUTHENTICATOR_REALM;
+import static org.sonar.db.loadedtemplate.LoadedTemplateDto.ONE_SHOT_TASK_TYPE;
+
+/**
+ * Feed users local property.
+ * If a realm is defined, then users are set as local only if their login are found in the property "sonar.security.localUsers",
+ * otherwise user are all set as local.
+ *
+ * See <a href="https://jira.sonarsource.com/browse/SONAR-7254">SONAR-7254</a>.
+ *
+ * Should be removed after LTS 5.X
+ *
+ * @since 5.5
+ */
+public class FeedUsersLocalStartupTask implements Startable {
+
+  private static final Logger LOG = Loggers.get(FeedUsersLocalStartupTask.class);
+
+  private static final String TEMPLATE_KEY = "UpdateUsersLocal";
+  private static final String LOCAL_USERS_PROPERTY = "sonar.security.localUsers";
+
+  private final System2 system2;
+
+  private final DbClient dbClient;
+  private final Settings settings;
+
+  public FeedUsersLocalStartupTask(System2 system2, DbClient dbClient, Settings settings) {
+    this.system2 = system2;
+    this.dbClient = dbClient;
+    this.settings = settings;
+  }
+
+  @Override
+  public void start() {
+    DbSession session = dbClient.openSession(false);
+    try {
+      if (hasAlreadyBeenExecuted(session)) {
+        return;
+      }
+      updateUsersLocal(session);
+      markAsExecuted(session);
+      session.commit();
+
+      if (settings.hasKey(LOCAL_USERS_PROPERTY)) {
+        LOG.info("NOTE : The property '{}' is now no more needed, you can safely remove it.", LOCAL_USERS_PROPERTY);
+      }
+    } finally {
+      dbClient.closeSession(session);
+    }
+  }
+
+  private void updateUsersLocal(DbSession session) {
+    long now = system2.now();
+    Set<String> localUsers = new HashSet<>(asList(settings.getStringArray(LOCAL_USERS_PROPERTY)));
+    boolean isRealmExist = settings.getString(CORE_AUTHENTICATOR_REALM) != null;
+    for (UserDto user : dbClient.userDao().selectUsers(session, UserQuery.ALL_ACTIVES)) {
```

Comment:
we don't need to update non active users? They can be reactivated, that's why I'm wondering